mrbean/src/main/java/com/fasterxml/jackson/module/mrbean/BeanBuilder.java

```diff
@@ -181,16 +181,28 @@ protected boolean hasConcreteOverride(Method m0, JavaType implementedType)
     {
         final String name = m0.getName();
         final Class<?>[] argTypes = m0.getParameterTypes();
+        try {
+            // getMethod returns the most-specific method implementation, for public methods only (which is any method in an interface)
+            Method effectiveMethod = implementedType.getRawClass().getMethod(name, argTypes);
+            if (BeanUtil.isConcrete(effectiveMethod)) {
```

Comment:
Quick question on this one: is the other case (not concrete) not conclusive?
That is, why not just return `BeanUtil.isConcrete(effectiveMethod)` as is, instead of continuing?
I am assuming here that if method is found, it is the visible declaration.

I suspect most of the time continuing would only be wasteful and is unlikely to give wrong answer....
but I have occasionally used pattern of re-abstract'ing methods (esp. `toString()` / `equals()` from `Object`) to force re-implementing by sub-classes.

Reply:
Oh, that's an interesting thought.  Yes... if the method is returned, then I think you have the definitive answer as to whether the method is concrete for the implemented type.

> I have occasionally used pattern of re-abstract'ing methods (esp. toString() / equals() from Object) to force re-implementing by sub-classes.
I think if the implemented type returned one of those "re-abstracted" methods, then it would be abstract, and the MrBean-generated class would be required to provide an implementation.  I'm not sure how MrBean would handle this, probably it would throw an error, since it's not a property?

So, to clarify what we're talking about, I think it would be correct to simply `return BeanUtil.isConcrete(effectiveMethod)` here.

Reply:
Ok cool.